crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -191,7 +191,12 @@ impl<T: TransactionOrdering> TxPool<T> {
 
     /// Returns all senders in the pool
     pub(crate) fn unique_senders(&self) -> HashSet<Address> {
-        self.all_transactions.txs.values().map(|tx| tx.transaction.sender()).collect()
+        self.all_transactions
+            .txs
+            .values()
+            .flat_map(|tx| tx.values())
+            .map(|tx| tx.transaction.sender())
```

Comment:
I think this now collects all tx senders but we only need to do this per value, so this would collect a bunch of redundant addresses, because we only need `tx.values().first()`